pkg/agent/manager/cache/cache.go

```diff
@@ -84,10 +121,24 @@ func (c *cacheImpl) Entry(regEntry *common.RegistrationEntry) *Entry {
 func (c *cacheImpl) SetEntry(entry *Entry) {
 	c.m.Lock()
 	defer c.m.Unlock()
+	subs := c.Subscribers.Get(entry.RegistrationEntry.Selectors)
+	for _, sub := range subs {
+		var entries []Entry
+		for _, e := range c.cache {
+			regEntrySelectors := selector.NewSetFromRaw(e[0].RegistrationEntry.Selectors)
+			if selector.NewSetFromRaw(sub.sel).IncludesSet(regEntrySelectors) {
+				entries = append(entries, e[0])
+			}
+		}
+		select {
+		case sub.C <- &WorkloadUpdate{cacheEntries: entries, bundle: c.bundle}:
+		default:
```

Comment:
will this default case trigger if the workload api is busy sending a previous update?